### ISSUE TITLE
Can't login with email address

### ISSUE THREAD

**Issue by titulebolide** [hedgedoc/hedgedoc#615]
**Description**
After updating from 1.6.0 to 1.7.0-cr1 (thanks to your merge #610 ), I managed to have the server migrating and runnning in production mode. Even if my database lloks fine (it don't seems to have been wiped), i'm unable to login with my email address.

**To Reproduce**
Update and migrate to 1.7.0-cr1 from 1.6.0 with a MariaDB database

**Logs**
There is absolutely no error logged, even if I am in debug mode. Here is what is logged after I try lo log in:
```
2020-12-02T20:02:59.640Z info: 	Executing (default): SELECT `id`, `profileid`, `profile`, `history`, `accessToken`, `refreshToken`, `deleteToken`, `email`, `password`, `createdAt`, `updatedAt` FROM `Users` AS `User` WHERE `User`.`email` = '************' LIMIT 1;
2020-12-02T20:03:00.584Z info: 	serializeUser: ***************************
2020-12-02T20:03:00.598Z info: 	Executing (default): SELECT `sid`, `expires`, `data`, `createdAt`, `updatedAt` FROM `Session` AS `Session` WHERE `Session`.`sid` = '**************************';
2020-12-02T20:03:00.616Z info: 	Executing (default): INSERT INTO `Session` (`sid`,`expires`,`data`,`createdAt`,`updatedAt`) VALUES (?,?,?,?,?);
2020-12-02T20:03:00.632Z info: 	82.64.116.9 - - [02/Dec/2020:20:03:00 +0000] "POST /login HTTP/1.1" 302 82 "https://codimd.**********.***/" "Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:83.0) Gecko/20100101 Firefox/83.0"

2020-12-02T20:03:00.835Z info: 	Executing (default): SELECT `sid`, `expires`, `data`, `createdAt`, `updatedAt` FROM `Session` AS `Session` WHERE `Session`.`sid` = '***************************';
2020-12-02T20:03:00.849Z info: 	Executing (default): INSERT INTO `Session` (`sid`,`expires`,`data`,`createdAt`,`updatedAt`) VALUES (?,?,?,?,?);
2020-12-02T20:03:00.867Z info: 	82.64.116.9 - - [02/Dec/2020:20:03:00 +0000] "GET / HTTP/1.1" 200 - "https://codimd.**********.***/" "Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:83.0) Gecko/20100101 Firefox/83.0"

2020-12-02T20:03:00.995Z info: 	Executing (default): SELECT `sid`, `expires`, `data`, `createdAt`, `updatedAt` FROM `Session` AS `Session` WHERE `Session`.`sid` = '******************************';
2020-12-02T20:03:01.008Z info: 	Executing (default): INSERT INTO `Session` (`sid`,`expires`,`data`,`createdAt`,`updatedAt`) VALUES (?,?,?,?,?);
2020-12-02T20:03:01.019Z info: 	82.64.116.9 - - [02/Dec/2020:20:03:01 +0000] "GET /config HTTP/1.1" 200 332 "https://codimd.**********.***/" "Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:83.0) Gecko/20100101 Firefox/83.0"

2020-12-02T20:03:01.143Z info: 	Executing (default): SELECT `sid`, `expires`, `data`, `createdAt`, `updatedAt` FROM `Session` AS `Session` WHERE `Session`.`sid` = '*********************************';
2020-12-02T20:03:01.160Z info: 	Executing (default): INSERT INTO `Session` (`sid`,`expires`,`data`,`createdAt`,`updatedAt`) VALUES (?,?,?,?,?);
2020-12-02T20:03:01.171Z info: 	82.64.116.9 - - [02/Dec/2020:20:03:01 +0000] "GET /me HTTP/1.1" 304 - "https://codimd.**********.***/" "Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:83.0) Gecko/20100101 Firefox/83.0"
```
And it stops there. It looks like it tries to get me a session four times in a row, and gives up.

**Config**

**Setup**
-    Host OS: Raspbian 10
-    NodeJS version: v12.18.4
-    HedgeDoc version (Click "Version info" in the footer)
     -   server 1.7.0-rc1
     -   client 1.7.0-rc1

Again, a huge thanks for Hedgedoc :heart:

**Comment by davidmehren** [comment]
Hmm, interesting.
Do you get a "Invalid email or password." error message when you try to login?
Do you get any errors in the browser console?

**Comment by titulebolide** [comment]
I have no error, nor from the client, neither from the browser console.

I added line 26 of lib/web/auth/email/index.js `logger.info(verified)` to see if the callback of verifyPassword was fired and if it is verified. It does, I got( a nice `info : true` in the console...

**Comment by titulebolide** [comment]
I tested it on my computer (not on my server) (fedora 33), and everything worked fine since it was a fresh install
I am going to try to copy the database

**Comment by titulebolide** [comment]
Wow it worked
The problem don't comes from the database itself...

**Comment by titulebolide** [comment]
Really strange... I had it working on my server without using the reverse-proxy (i ssh-ed to my server and connected to the localhost via SOCKS5), and the auth worked. Maybe the callbacks from the server can't get through?
I'm investigating

**Comment by titulebolide** [comment]
Okay.... I iterated from what worked in my previous comment, i restaured the proxy without ssl, my browser wasn't happy because in its cache it was supposed to be ssl, so a switched for private session, and it worked. From that i restaured ssl. And i kept working....

Thus I'm closing, but it is really strange, I changed nothing. I'm thinking it might be the browser's cache in some way? My sessionSecret was not set, so maybe he wasn't happy about it... ?

Sorry for having bothered you ;)
